NEONLidarPipeline.py

```diff
@@ -0,0 +1,186 @@
+class NEONLidarPipeline:
+    """
+    A class to execute a pipeline for NEON lidar data.
+
+    Attributes:
+        site_name (str): The name of the NEON site.
+        separator (str): The separator used in the insitu data file.
+        base_dir (str): The base directory for the site data.
+        insitu_path (str): The path to the insitu vegetation structure data.
+        chm_path (str): The path to the lidar canopy height (CHM) data.
+        plot_path (str): The path to the plot centroid shapefile.
+    """
+    import os
+    import pandas as pd
+    import geopandas as gpd
+    import rasterstats as rs
+
+    def __init__(self, site_name: str, separator: str):
+        """
+        Initialize the NEONLidarPipeline class.
```

Comment:
@daltonkw I don't think you need to write a doc string for the __init__ function, although this is very descriptive!